test/cmdlineargs.jl

```diff
@@ -256,45 +256,63 @@ let exename = `$(Base.julia_cmd()) --precompiled=yes --startup-file=no`
             # write a julia source file that just prints ARGS to STDOUT
             write(testfile, """
                 println(ARGS)
-            """)
+                """
+            )
             @test readchomp(`$exename $testfile foo -bar --baz`) ==
                 "String[\"foo\", \"-bar\", \"--baz\"]"
             @test readchomp(`$exename $testfile -- foo -bar --baz`) ==
                 "String[\"foo\", \"-bar\", \"--baz\"]"
             @test readchomp(`$exename -L $testfile -e 'exit(0)' -- foo -bar --baz`) ==
                 "String[\"foo\", \"-bar\", \"--baz\"]"
             @test split(readchomp(`$exename -L $testfile $testfile`), '\n') ==
-                ["String[\"$(escape(testfile))\"]", "String[]"]
+                ["String[]", "String[]"]
             @test !success(`$exename --foo $testfile`)
             @test readchomp(`$exename -L $testfile -e 'exit(0)' -- foo -bar -- baz`) == "String[\"foo\", \"-bar\", \"--\", \"baz\"]"
         finally
             rm(testfile)
         end
     end
 
-    # test the script name
-    let a = tempname(), b = tempname()
-        try
-            write(a, """
-                println(@__FILE__)
-                println(PROGRAM_FILE)
-                println(length(ARGS))
-                include(\"$(escape(b))\")
-            """)
-            write(b, """
-                println(@__FILE__)
-                println(PROGRAM_FILE)
-                println(length(ARGS))
-            """)
-            @test split(readchomp(`$exename $a`), '\n') ==
-                ["$a", "$a", "0", "$b", "$a", "0"]
-            @test split(readchomp(`$exename -L $b -e 'exit(0)'`), '\n') ==
-                ["$(realpath(b))", "", "0"]
-            @test split(readchomp(`$exename -L $b $a`), '\n') ==
-                ["$(realpath(b))", "", "1", "$a", "$a", "0", "$b", "$a", "0"]
-        finally
-            rm(a)
-            rm(b)
+    # test the program name remains constant
+    mktempdir() do dir
+        a = joinpath(dir, "a.jl")
+        b = joinpath(dir, "b.jl")
+        c = joinpath(dir, ".juliarc.jl")
+
+        write(a, """
+            println(@__FILE__)
+            println(PROGRAM_FILE)
+            include(\"$(escape(b))\")
+            """
+        )
```

Comment:
nitpick, but I don't think it's usually worth using up vertical space to put close parens on their own line